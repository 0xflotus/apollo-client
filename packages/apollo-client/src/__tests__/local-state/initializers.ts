import { InMemoryCache } from 'apollo-cache-inmemory';
import { ApolloLink } from 'apollo-link';
import gql from 'graphql-tag';

import ApolloClient from '../..';

describe('General use', () => {
  it('should write the result of initiailizer functions to the cache', () => {
    const cache = new InMemoryCache();
    new ApolloClient({
      cache,
      link: ApolloLink.empty(),
      initializers: {
        foo: () => ({ bar: false, __typename: 'Bar' }),
      },
      resolvers: {
        Query: {
          foo: () => ({ bar: true }),
        },
      },
    });
    expect(cache.extract()).toMatchSnapshot();
  });

  it('should not call the resolver if the data is already in the cache', () => {
    const fooResolver = jest.fn();
    const resolvers = { Query: { foo: fooResolver } };

    const cache = new InMemoryCache();
    const client = new ApolloClient({
      cache,
      link: ApolloLink.empty(),
      initializers: {
        foo: () => ({ bar: false, __typename: 'Bar' }),
      },
      resolvers,
    });

    const query = gql`
      {
        foo @client {
          bar
        }
      }
    `;

    client
      .query({ query })
      .then(({ data }) => {
        expect(fooResolver).not.toHaveBeenCalled();
      })
      .catch(e => console.error(e));
  });
});
