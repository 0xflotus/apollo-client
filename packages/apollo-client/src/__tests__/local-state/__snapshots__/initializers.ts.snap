// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General use should write the result of initiailizer functions to the cache 1`] = `
Object {
  "$ROOT_QUERY.foo": Object {
    "__typename": "Bar",
    "bar": false,
  },
  "ROOT_QUERY": Object {
    "foo": Object {
      "generated": true,
      "id": "$ROOT_QUERY.foo",
      "type": "id",
      "typename": "Bar",
    },
  },
}
`;
